shop-service/import-service/handlers/importFileParser.ts

```diff
@@ -0,0 +1,62 @@
+import { Context, S3Event } from "aws-lambda";
+import * as AWS from "aws-sdk";
+import { ReasonPhrases, StatusCodes } from "http-status-codes";
+import * as csv from "csv-parser";
+import * as dotenv from 'dotenv';
+
+import createResponse from "../helpers/createResponse";
+
+dotenv.config();
+
+export const importFileParser = async (event: S3Event, context: Context): Promise<any> => {
+  console.log(
+    `Event:\n ${JSON.stringify(event)}\nContext:\n${JSON.stringify(context)}`
+  );
+  try {
+    const { BUCKET } = process.env;
+    const s3 = new AWS.S3({ region: "eu-west-1" });
+
+    await new Promise((resolve, reject) => {
+      event.Records.forEach((record) => {
```

Comment:
In the case of several files are uploaded at once, you will resolve the Promise after the first processed file.
From the async/await point of view, you should create a Promise for each record and wait for all of them are resolved.

Reply:
thanks for your comment! Fixed it